test/unittests/testutil.cc

```diff
@@ -245,7 +247,55 @@ MockCatalog* MockCatalog::AttachFreely(const std::string  &root_path,
   return new_catalog;
 }
 
-void MockCatalog::RegisterChild(MockCatalog *child) {
+void MockCatalog::RemoveChild(MockCatalog *child) {
```

Comment:
In normal catalogs the information about direct nested catalogs is stored in the database, so that you can easily know what is the total list of children-catalog. Also, catalogs have a list of already attached children, which is always a subset of the ones mentioned before.

In the mock catalog there is sqlite database, so there are two lists: one for the available children and another for the mounted children. The first one is constant, but the second one can be modified. In any case it's necessary to have different methods to operate with each list, and that includes inserting and, only in the mount list, deleting elements.